lib/spack/spack/build_systems/cuda.py

```diff
@@ -95,14 +95,16 @@ def cuda_flags(arch_list):
     conflicts('%gcc@9:', when='+cuda ^cuda@:10.2.89' + arch_platform)
     conflicts('%gcc@:4', when='+cuda ^cuda@11.0.2:' + arch_platform)
     conflicts('%gcc@10:', when='+cuda ^cuda@:11.0.2' + arch_platform)
+    conflicts('%gcc@11:', when='+cuda ^cuda@:11.1.0' + arch_platform)
     conflicts('%pgi@:14.8', when='+cuda ^cuda@:7.0.27' + arch_platform)
     conflicts('%pgi@:15.3,15.5:', when='+cuda ^cuda@7.5' + arch_platform)
     conflicts('%pgi@:16.2,16.0:16.3', when='+cuda ^cuda@8' + arch_platform)
     conflicts('%pgi@:15,18:', when='+cuda ^cuda@9.0:9.1' + arch_platform)
     conflicts('%pgi@:16,19:', when='+cuda ^cuda@9.2.88:10' + arch_platform)
     conflicts('%pgi@:17,20:',
               when='+cuda ^cuda@10.1.105:10.2.89' + arch_platform)
-    conflicts('%pgi@:17,20.2:', when='+cuda ^cuda@11.0.2' + arch_platform)
+    conflicts('%pgi@:17,20.2:',
```

Comment:
According to the headers, this is all that is checked (don't trust the compiler table, bug reports opened)
```suggestion
    conflicts('%pgi@:17,21:',
```